source/slang/slang-check-overload.cpp

```diff
@@ -1561,6 +1547,22 @@ namespace Slang
             auto overloadRankDiff = getOverloadRank(right->item.declRef) - getOverloadRank(left->item.declRef);
             if (overloadRankDiff)
                 return overloadRankDiff;
+
+            // We need to consider the distance of the declarations to the global scope to resolve this case:
```

Comment:
This need to be before `overloadRank` comparison above. Overload rank is a hack that we should always use as last resort disambiguation method.

Reply:
ACK.